notes_model_v2/parameters_search.md
### Parameters #1



{

&#x20;   "N": {

&#x20;       "BG\_dl": 2,

&#x20;       "BG\_dm": 2,

&#x20;       "BG\_v": 2,

&#x20;       "MGV": 2,

&#x20;       "MC": 2,

&#x20;       "BLA\_IC": 4,

&#x20;       "SNpc": 2,

&#x20;       "PPN": 1,

&#x20;       "LH": 1,

&#x20;       "VTA": 1,

&#x20;       "P": 2,

&#x20;       "DM": 2,

&#x20;       "PL": 2,

&#x20;       "PFCd\_PPC": 2

&#x20;   },

&#x20;   "threshold": {

&#x20;       "BG\_dl": 0.0,

&#x20;       "BG\_dm": 0.0,

&#x20;       "BG\_v": 0.0,

&#x20;       "MGV": 0.0,

&#x20;       "MC": 0.4,

&#x20;       "BLA\_IC": 0.0,

&#x20;       "SNpc": 0.0,

&#x20;       "PPN": 0.0,

&#x20;       "LH": 0.0,

&#x20;       "VTA": 0.0,

&#x20;       "P": 0.0,

&#x20;       "DM": 0.0,

&#x20;       "PL": 0.0,

&#x20;       "PFCd\_PPC": 0.0

&#x20;   },

&#x20;   "tau": {

&#x20;       "MC": 6,

&#x20;       "PFCd\_PPC": 6,

&#x20;       "PL": 6,

&#x20;       "MGV": 6,

&#x20;       "P": 6,

&#x20;       "DM": 6,

&#x20;       "BG\_dl": 6,

&#x20;       "BG\_dm": 6,

&#x20;       "BG\_v": 6,

&#x20;       "BLA\_IC": \[

&#x20;           2,

&#x20;           10

&#x20;       ],

&#x20;       "SNpc\_i": 6,

&#x20;       "SNpc\_o": 6,

&#x20;       "PPN": \[

&#x20;           2,

&#x20;           10

&#x20;       ],

&#x20;       "LH": \[

&#x20;           2,

&#x20;           10

&#x20;       ],

&#x20;       "VTA": 6

&#x20;   },

&#x20;   "baseline": {

&#x20;       "PPN": 0.0,

&#x20;       "LH": 0.0,

&#x20;       "VTA": 0.2,

&#x20;       "BLA\_IC": 0.0,

&#x20;       "SNpci": 2.0,

&#x20;       "SNpco": 0.7,

&#x20;       "GPe": 0.33,

&#x20;       "DLS\_1": 0.2,

&#x20;       "DLS\_2": 0.2,

&#x20;       "STNdl": 0.2,

&#x20;       "GPi": 0.33,

&#x20;       "DMS\_1": 0.0,

&#x20;       "DMS\_2": 0.0,

&#x20;       "STNdm": 0.0,

&#x20;       "GPi\_SNpr": 0.33,

&#x20;       "NAc\_1": 0.0,

&#x20;       "NAc\_2": 0.0,

&#x20;       "STNv": 0.0,

&#x20;       "SNpr": 0.33,

&#x20;       "MGV": 0.2,

&#x20;       "MC": 0.0,

&#x20;       "P": 0.2,

&#x20;       "PFCd\_PPC": 0.0,

&#x20;       "DM": 0.2,

&#x20;       "PL": 0.0

&#x20;   },

&#x20;   "BG\_dl\_W": {

&#x20;       "DLS\_1\_GPi\_W": 3.0,

&#x20;       "DLS\_2\_GPe\_W": 2.5,

&#x20;       "STNdl\_GPi\_W": 3.0,

&#x20;       "STNdl\_GPe\_W": 2.0,

&#x20;       "GPe\_STNdl\_W": 1.5,

&#x20;       "GPe\_GPi\_W": 0.1

&#x20;   },

&#x20;   "BG\_dm\_W": {

&#x20;       "DMS\_1\_GPi\_SNpr\_W": 3.0,

&#x20;       "DMS\_2\_GPe\_W": 2.5,

&#x20;       "STNdm\_GPi\_SNpr\_W": 3.0,

&#x20;       "STNdm\_GPe\_W": 2.0,

&#x20;       "GPe\_STNdm\_W": 1.5,

&#x20;       "GPe\_GPi\_SNpr\_W": 0.1

&#x20;   },

&#x20;   "BG\_v\_W": {

&#x20;       "NAc\_1\_SNpr\_W": 3.0,

&#x20;       "NAc\_2\_GPe\_W": 2.5,

&#x20;       "STNv\_SNpr\_W": 3.0,

&#x20;       "STNv\_GPe\_W": 2.0,

&#x20;       "GPe\_STNv\_W": 1.5,

&#x20;       "GPe\_SNpr\_W": 0.1

&#x20;   },

&#x20;   "SNpc\_W": {

&#x20;       "SNpci\_1\_SNpco\_1\_W": 0.54,

&#x20;       "SNpci\_2\_SNpco\_2\_W": 0.54

&#x20;   },

&#x20;   "seed": 0,

&#x20;   "noise": {

&#x20;       "BG\_dl": 0.0,

&#x20;       "BG\_dm": 0.0,

&#x20;       "BG\_v": 0.0,

&#x20;       "MGV": 0.0,

&#x20;       "MC": 0.4,

&#x20;       "BLA\_IC": 0.0,

&#x20;       "SNpc": 0.0,

&#x20;       "PPN": 0.0,

&#x20;       "LH": 0.0,

&#x20;       "VTA": 0.0,

&#x20;       "P": 0.0,

&#x20;       "DM": 0.0,

&#x20;       "PL": 0.4,

&#x20;       "PFCd\_PPC": 0.4

&#x20;   },

&#x20;   "Matrices\_scalars": {

&#x20;       "Mani\_DLS": 0.0,

&#x20;       "Mani\_DMS": 0.0,

&#x20;       "Mani\_BLA\_IC": 5.0,

&#x20;       "Food\_PPN": 10.0,

&#x20;       "Food\_BLA\_IC": 5.0,

&#x20;       "Food\_LH": 10.0,

&#x20;       "Sat\_BLA\_IC": 20.0,

&#x20;       "PPN\_SNpco": 5.0,

&#x20;       "BLA\_IC\_NAc": 0.0,

&#x20;       "BLA\_IC\_LH": 5.0,

&#x20;       "LH\_VTA": 20.0,

&#x20;       "NAc\_1\_SNpci\_1": 4.0,

&#x20;       "DMS\_1\_SNpci\_2": 4.0,

&#x20;       "GPi\_MGV": 1.5,

&#x20;       "GPi\_SNpr\_P": 1.5,

&#x20;       "SNpr\_DM": 1.5,

&#x20;       "MGV\_MC": 1.5,

&#x20;       "P\_PFCd\_PPC": 1.5,

&#x20;       "DM\_PL": 1.5,

&#x20;       "PL\_DM": 2.4,

&#x20;       "PL\_NAc\_1": 0.9,

&#x20;       "PL\_NAc\_2": 0.8,

&#x20;       "PL\_STNv": 1.4,

&#x20;       "PL\_PFCd\_PPC": 0.3,

&#x20;       "PFCd\_PPC\_P": 2.4,

&#x20;       "PFCd\_PPC\_DMS\_1": 0.9,

&#x20;       "PFCd\_PPC\_DMS\_2": 0.8,

&#x20;       "PFCd\_PPC\_STNdm": 1.4,

&#x20;       "PFCd\_PPC\_PL": 0.2,

&#x20;       "PFCd\_PPC\_MC": 0.2,

&#x20;       "MC\_MGV": 2.4,

&#x20;       "MC\_DLS\_1": 0.9,

&#x20;       "MC\_DLS\_2": 0.8,

&#x20;       "MC\_STNdl": 1.4,

&#x20;       "MC\_PFCd\_PPC": 0.3

&#x20;   },

&#x20;   "DA\_values": {

&#x20;       "Y\_DLS\_1": 0.1,

&#x20;       "Y\_DLS\_2": 0.2,

&#x20;       "Y\_DMS\_1": 0.1,

&#x20;       "Y\_DMS\_2": 0.2,

&#x20;       "Y\_NAc\_1": 0.1,

&#x20;       "Y\_NAc\_2": 0.2,

&#x20;       "delta\_DLS\_1": 0.5,

&#x20;       "delta\_DLS\_2": 5.0,

&#x20;       "delta\_DMS\_1": 0.5,

&#x20;       "delta\_DMS\_2": 5.0,

&#x20;       "delta\_NAc\_1": 0.5,

&#x20;       "delta\_NAc\_2": 5.0

&#x20;   },

&#x20;   "BLA\_Learn": {

&#x20;       "eta\_b": 0.2,

&#x20;       "alpha\_t": 50.0,

&#x20;       "tau\_t": 80,

&#x20;       "theta\_DA": 0.5,

&#x20;       "max\_W": 2

&#x20;   },

&#x20;   "Str\_Learn": {

&#x20;       "eta\_DLS\_1": 0.04,

&#x20;       "eta\_DLS\_2": 0.04,

&#x20;       "eta\_DMS\_1": 0.04,

&#x20;       "eta\_DMS\_2": 0.04,

&#x20;       "eta\_NAc\_1": 0.1,

&#x20;       "eta\_NAc\_2": 0.1,

&#x20;       "theta\_DA\_DLS\_1": 0.5,

&#x20;       "theta\_DA\_DLS\_2": 0.2,

&#x20;       "theta\_DA\_DMS\_1": 0.5,

&#x20;       "theta\_DA\_DMS\_2": 0.2,

&#x20;       "theta\_DA\_NAc\_1": 0.5,

&#x20;       "theta\_DA\_NAc\_2": 0.2,

&#x20;       "theta\_DLS\_1": 0.45,

&#x20;       "theta\_DLS\_2": 0.8,

&#x20;       "theta\_DMS\_1": 0.45,

&#x20;       "theta\_DMS\_2": 0.8,

&#x20;       "theta\_NAc\_1": 0.45,

&#x20;       "theta\_NAc\_2": 0.8,

&#x20;       "theta\_inp\_DLS\_1": 0.5,

&#x20;       "theta\_inp\_DLS\_2": 0.5,

&#x20;       "theta\_inp\_DMS\_1": 0.5,

&#x20;       "theta\_inp\_DMS\_2": 0.5,

&#x20;       "theta\_inp\_NAc\_1": 0.5,

&#x20;       "theta\_inp\_NAc\_2": 0.5,

&#x20;       "max\_W\_DLS": 1,

&#x20;       "max\_W\_DMS": 1,

&#x20;       "max\_W\_NAc": 2

&#x20;   }

}



* Model behaves properly in single module simulation, need to test BGv and Amigdala modules



### Parameters #2



{

&#x20;   "N": {

&#x20;       "BG\_dl": 2,

&#x20;       "BG\_dm": 2,

&#x20;       "BG\_v": 2,

&#x20;       "MGV": 2,

&#x20;       "MC": 2,

&#x20;       "BLA\_IC": 4,

&#x20;       "SNpc": 1,

&#x20;       "PPN": 1,

&#x20;       "LH": 1,

&#x20;       "VTA": 1,

&#x20;       "P": 2,

&#x20;       "DM": 2,

&#x20;       "PL": 2,

&#x20;       "PFCd\_PPC": 2

&#x20;   },

&#x20;   "threshold": {

&#x20;       "BG\_dl": 0.0,

&#x20;       "BG\_dm": 0.0,

&#x20;       "BG\_v": 0.0,

&#x20;       "MGV": 0.0,

&#x20;       "MC": 0.4,

&#x20;       "BLA\_IC": 0.0,

&#x20;       "SNpc": 0.0,

&#x20;       "PPN": 0.0,

&#x20;       "LH": 0.0,

&#x20;       "VTA": 0.0,

&#x20;       "P": 0.0,

&#x20;       "DM": 0.0,

&#x20;       "PL": 0.0,

&#x20;       "PFCd\_PPC": 0.0

&#x20;   },

&#x20;   "tau": {

&#x20;       "MC": 6,

&#x20;       "PFCd\_PPC": 6,

&#x20;       "PL": 6,

&#x20;       "MGV": 6,

&#x20;       "P": 6,

&#x20;       "DM": 6,

&#x20;       "BG\_dl": 6,

&#x20;       "BG\_dm": 6,

&#x20;       "BG\_v": 6,

&#x20;       "BLA\_IC": \[

&#x20;           2,

&#x20;           10

&#x20;       ],

&#x20;       "SNpc\_i": 6,

&#x20;       "SNpc\_o": 6,

&#x20;       "PPN": \[

&#x20;           2,

&#x20;           10

&#x20;       ],

&#x20;       "LH": \[

&#x20;           2,

&#x20;           10

&#x20;       ],

&#x20;       "VTA": 6

&#x20;   },

&#x20;   "baseline": {

&#x20;       "PPN": 0.0,

&#x20;       "LH": 0.0,

&#x20;       "VTA": 0.3,

&#x20;       "BLA\_IC": 0.0,

&#x20;       "SNpci": 2.0,

&#x20;       "SNpco": 0.7,

&#x20;       "GPe": 0.33,

&#x20;       "DLS\_1": 0.2,

&#x20;       "DLS\_2": 0.2,

&#x20;       "STNdl": 0.2,

&#x20;       "GPi": 0.33,

&#x20;       "DMS\_1": 0.0,

&#x20;       "DMS\_2": 0.0,

&#x20;       "STNdm": 0.0,

&#x20;       "GPi\_SNpr": 0.33,

&#x20;       "NAc\_1": 0.0,

&#x20;       "NAc\_2": 0.0,

&#x20;       "STNv": 0.0,

&#x20;       "SNpr": 0.33,

&#x20;       "MGV": 0.2,

&#x20;       "MC": 0.0,

&#x20;       "P": 0.2,

&#x20;       "PFCd\_PPC": 0.0,

&#x20;       "DM": 0.2,

&#x20;       "PL": 0.0

&#x20;   },

&#x20;   "BG\_dl\_W": {

&#x20;       "DLS\_1\_GPi\_W": 3.0,

&#x20;       "DLS\_2\_GPe\_W": 2.5,

&#x20;       "STNdl\_GPi\_W": 3.0,

&#x20;       "STNdl\_GPe\_W": 2.0,

&#x20;       "GPe\_STNdl\_W": 1.5,

&#x20;       "GPe\_GPi\_W": 0.1

&#x20;   },

&#x20;   "BG\_dm\_W": {

&#x20;       "DMS\_1\_GPi\_SNpr\_W": 3.0,

&#x20;       "DMS\_2\_GPe\_W": 2.5,

&#x20;       "STNdm\_GPi\_SNpr\_W": 3.0,

&#x20;       "STNdm\_GPe\_W": 2.0,

&#x20;       "GPe\_STNdm\_W": 1.5,

&#x20;       "GPe\_GPi\_SNpr\_W": 0.1

&#x20;   },

&#x20;   "BG\_v\_W": {

&#x20;       "NAc\_1\_SNpr\_W": 3.0,

&#x20;       "NAc\_2\_GPe\_W": 2.5,

&#x20;       "STNv\_SNpr\_W": 3.0,

&#x20;       "STNv\_GPe\_W": 2.0,

&#x20;       "GPe\_STNv\_W": 1.5,

&#x20;       "GPe\_SNpr\_W": 0.1

&#x20;   },

&#x20;   "SNpc\_W": {

&#x20;       "SNpci\_1\_SNpco\_1\_W": 0.54,

&#x20;       "SNpci\_2\_SNpco\_2\_W": 0.54

&#x20;   },

&#x20;   "seed": 0,

&#x20;   "noise": {

&#x20;       "BG\_dl": 0.0,

&#x20;       "BG\_dm": 0.0,

&#x20;       "BG\_v": 0.0,

&#x20;       "MGV": 0.0,

&#x20;       "MC": 0.4,

&#x20;       "BLA\_IC": 0.0,

&#x20;       "SNpc": 0.0,

&#x20;       "PPN": 0.0,

&#x20;       "LH": 0.0,

&#x20;       "VTA": 0.0,

&#x20;       "P": 0.0,

&#x20;       "DM": 0.0,

&#x20;       "PL": 0.4,

&#x20;       "PFCd\_PPC": 0.4

&#x20;   },

&#x20;   "Matrices\_scalars": {

&#x20;       "Mani\_DLS": 0.0,

&#x20;       "Mani\_DMS": 0.0,

&#x20;       "Mani\_BLA\_IC": 5.0,

&#x20;       "Food\_PPN": 10.0,

&#x20;       "Food\_BLA\_IC": 5.0,

&#x20;       "Food\_LH": 10.0,

&#x20;       "Sat\_BLA\_IC": 20.0,

&#x20;       "PPN\_SNpco": 5.0,

&#x20;       "BLA\_IC\_NAc": 0.0,

&#x20;       "BLA\_IC\_LH": 5.0,

&#x20;       "LH\_VTA": 20.0,

&#x20;       "NAc\_1\_SNpci\_1": 4.0,

&#x20;       "DMS\_1\_SNpci\_2": 4.0,

&#x20;       "GPi\_MGV": 1.5,

&#x20;       "GPi\_SNpr\_P": 1.5,

&#x20;       "SNpr\_DM": 1.5,

&#x20;       "MGV\_MC": 1.5,

&#x20;       "P\_PFCd\_PPC": 1.5,

&#x20;       "DM\_PL": 1.5,

&#x20;       "PL\_DM": 2.4,

&#x20;       "PL\_NAc\_1": 1.2,

&#x20;       "PL\_NAc\_2": 1.0,

&#x20;       "PL\_STNv": 1.4,

&#x20;       "PL\_PFCd\_PPC": 0.1,

&#x20;       "PFCd\_PPC\_P": 2.4,

&#x20;       "PFCd\_PPC\_DMS\_1": 0.9,

&#x20;       "PFCd\_PPC\_DMS\_2": 0.8,

&#x20;       "PFCd\_PPC\_STNdm": 1.4,

&#x20;       "PFCd\_PPC\_PL": 0.2,

&#x20;       "PFCd\_PPC\_MC": 0.2,

&#x20;       "MC\_MGV": 2.4,

&#x20;       "MC\_DLS\_1": 0.9,

&#x20;       "MC\_DLS\_2": 0.8,

&#x20;       "MC\_STNdl": 1.4,

&#x20;       "MC\_PFCd\_PPC": 0.1

&#x20;   },

&#x20;   "DA\_values": {

&#x20;       "Y\_DLS\_1": 0.1,

&#x20;       "Y\_DLS\_2": 0.2,

&#x20;       "Y\_DMS\_1": 0.1,

&#x20;       "Y\_DMS\_2": 0.2,

&#x20;       "Y\_NAc\_1": 0.1,

&#x20;       "Y\_NAc\_2": 0.2,

&#x20;       "delta\_DLS\_1": 0.5,

&#x20;       "delta\_DLS\_2": 5.0,

&#x20;       "delta\_DMS\_1": 0.5,

&#x20;       "delta\_DMS\_2": 5.0,

&#x20;       "delta\_NAc\_1": 0.5,

&#x20;       "delta\_NAc\_2": 5.0

&#x20;   },

&#x20;   "BLA\_Learn": {

&#x20;       "eta\_b": 0.2,

&#x20;       "alpha\_t": 50.0,

&#x20;       "tau\_t": 80,

&#x20;       "theta\_DA": 0.5,

&#x20;       "max\_W": 2

&#x20;   },

&#x20;   "Str\_Learn": {

&#x20;       "eta\_DLS\_1": 0.04,

&#x20;       "eta\_DLS\_2": 0.04,

&#x20;       "eta\_DMS\_1": 0.04,

&#x20;       "eta\_DMS\_2": 0.04,

&#x20;       "eta\_NAc\_1": 0.1,

&#x20;       "eta\_NAc\_2": 0.1,

&#x20;       "theta\_DA\_DLS\_1": 0.5,

&#x20;       "theta\_DA\_DLS\_2": 0.3,

&#x20;       "theta\_DA\_DMS\_1": 0.5,

&#x20;       "theta\_DA\_DMS\_2": 0.3,

&#x20;       "theta\_DA\_NAc\_1": 0.5,

&#x20;       "theta\_DA\_NAc\_2": 0.3,

&#x20;       "theta\_DLS\_1": 0.45,

&#x20;       "theta\_DLS\_2": 0.8,

&#x20;       "theta\_DMS\_1": 0.45,

&#x20;       "theta\_DMS\_2": 0.8,

&#x20;       "theta\_NAc\_1": 0.45,

&#x20;       "theta\_NAc\_2": 0.8,

&#x20;       "theta\_inp\_DLS\_1": 0.5,

&#x20;       "theta\_inp\_DLS\_2": 0.5,

&#x20;       "theta\_inp\_DMS\_1": 0.5,

&#x20;       "theta\_inp\_DMS\_2": 0.5,

&#x20;       "theta\_inp\_NAc\_1": 0.5,

&#x20;       "theta\_inp\_NAc\_2": 0.5,

&#x20;       "max\_W\_DLS": 1,

&#x20;       "max\_W\_DMS": 1,

&#x20;       "max\_W\_NAc": 2

&#x20;   }

}



* First testing with entire model, 10 trials. Results are discussable but not that bad. Can proceed from here to test a potential healthy simulation



### Parameters #3



"Str\_Learn": {

&#x20;       "eta\_DLS\_1": 0.04,

&#x20;       "eta\_DLS\_2": 0.04,

&#x20;       "eta\_DMS\_1": 0.04,

&#x20;       "eta\_DMS\_2": 0.04,

&#x20;       "eta\_NAc\_1": 0.1,

&#x20;       "eta\_NAc\_2": 0.1,

&#x20;       "theta\_DA\_DLS\_1": 0.5,

&#x20;       "theta\_DA\_DLS\_2": 0.5,

&#x20;       "theta\_DA\_DMS\_1": 0.5,

&#x20;       "theta\_DA\_DMS\_2": 0.5,

&#x20;       "theta\_DA\_NAc\_1": 0.5,

&#x20;       "theta\_DA\_NAc\_2": 0.5,

&#x20;       "theta\_DLS\_1": 0.45,

&#x20;       "theta\_DLS\_2": 0.2,

&#x20;       "theta\_DMS\_1": 0.45,

&#x20;       "theta\_DMS\_2": 0.2,

&#x20;       "theta\_NAc\_1": 0.45,

&#x20;       "theta\_NAc\_2": 0.2,

&#x20;       "theta\_inp\_DLS\_1": 0.5,

&#x20;       "theta\_inp\_DLS\_2": 0.5,

&#x20;       "theta\_inp\_DMS\_1": 0.5,

&#x20;       "theta\_inp\_DMS\_2": 0.5,

&#x20;       "theta\_inp\_NAc\_1": 0.5,

&#x20;       "theta\_inp\_NAc\_2": 0.5,

&#x20;       "max\_W\_DLS": 1,

&#x20;       "max\_W\_DMS": 1,

&#x20;       "max\_W\_NAc": 2

&#x20;   }



* Changing in the Str\_Learn parameters for D2 learning after learning function adaptation: **ΔW∝AB−λXOR(A,B)**



### **Parameters #4**



**{**

&#x20;   **"N": {**

&#x20;       **"BG\_dl": 2,**

&#x20;       **"BG\_dm": 2,**

&#x20;       **"BG\_v": 2,**

&#x20;       **"MGV": 2,**

&#x20;       **"MC": 2,**

&#x20;       **"BLA\_IC": 4,**

&#x20;       **"SNpc": 1,**

&#x20;       **"PPN": 1,**

&#x20;       **"LH": 1,**

&#x20;       **"VTA": 1,**

&#x20;       **"P": 2,**

&#x20;       **"DM": 2,**

&#x20;       **"PL": 2,**

&#x20;       **"PFCd\_PPC": 2**

&#x20;   **},**

&#x20;   **"threshold": {**

&#x20;       **"BG\_dl": 0.0,**

&#x20;       **"BG\_dm": 0.0,**

&#x20;       **"BG\_v": 0.0,**

&#x20;       **"MGV": 0.0,**

&#x20;       **"MC": 0.5,**

&#x20;       **"BLA\_IC": 0.0,**

&#x20;       **"SNpc": 0.0,**

&#x20;       **"PPN": 0.0,**

&#x20;       **"LH": 0.0,**

&#x20;       **"VTA": 0.0,**

&#x20;       **"P": 0.0,**

&#x20;       **"DM": 0.0,**

&#x20;       **"PL": 0.0,**

&#x20;       **"PFCd\_PPC": 0.5**

&#x20;   **},**

&#x20;   **"tau": {**

&#x20;       **"MC": 6,**

&#x20;       **"PFCd\_PPC": 6,**

&#x20;       **"PL": 6,**

&#x20;       **"MGV": 6,**

&#x20;       **"P": 6,**

&#x20;       **"DM": 6,**

&#x20;       **"BG\_dl": 6,**

&#x20;       **"BG\_dm": 6,**

&#x20;       **"BG\_v": 6,**

&#x20;       **"BLA\_IC": \[**

&#x20;           **2,**

&#x20;           **10**

&#x20;       **],**

&#x20;       **"SNpc\_i": 6,**

&#x20;       **"SNpc\_o": 6,**

&#x20;       **"PPN": \[**

&#x20;           **2,**

&#x20;           **10**

&#x20;       **],**

&#x20;       **"LH": \[**

&#x20;           **2,**

&#x20;           **10**

&#x20;       **],**

&#x20;       **"VTA": 6**

&#x20;   **},**

&#x20;   **"baseline": {**

&#x20;       **"PPN": 0.0,**

&#x20;       **"LH": 0.0,**

&#x20;       **"VTA": 0.45,**

&#x20;       **"BLA\_IC": 0.0,**

&#x20;       **"SNpci": 2.0,**

&#x20;       **"SNpco": 0.7,**

&#x20;       **"GPe": 0.33,**

&#x20;       **"DLS\_1": 0.2,**

&#x20;       **"DLS\_2": 0.2,**

&#x20;       **"STNdl": 0.2,**

&#x20;       **"GPi": 0.33,**

&#x20;       **"DMS\_1": 0.2,**

&#x20;       **"DMS\_2": 0.2,**

&#x20;       **"STNdm": 0.2,**

&#x20;       **"GPi\_SNpr": 0.33,**

&#x20;       **"NAc\_1": 0.2,**

&#x20;       **"NAc\_2": 0.2,**

&#x20;       **"STNv": 0.2,**

&#x20;       **"SNpr": 0.33,**

&#x20;       **"MGV": 0.2,**

&#x20;       **"MC": 0.0,**

&#x20;       **"P": 0.2,**

&#x20;       **"PFCd\_PPC": 0.0,**

&#x20;       **"DM": 0.2,**

&#x20;       **"PL": 0.0**

&#x20;   **},**

&#x20;   **"BG\_dl\_W": {**

&#x20;       **"DLS\_1\_GPi\_W": 3.0,**

&#x20;       **"DLS\_2\_GPe\_W": 2.5,**

&#x20;       **"STNdl\_GPi\_W": 3.0,**

&#x20;       **"STNdl\_GPe\_W": 2.0,**

&#x20;       **"GPe\_STNdl\_W": 1.5,**

&#x20;       **"GPe\_GPi\_W": 0.1**

&#x20;   **},**

&#x20;   **"BG\_dm\_W": {**

&#x20;       **"DMS\_1\_GPi\_SNpr\_W": 3.0,**

&#x20;       **"DMS\_2\_GPe\_W": 2.5,**

&#x20;       **"STNdm\_GPi\_SNpr\_W": 3.0,**

&#x20;       **"STNdm\_GPe\_W": 2.0,**

&#x20;       **"GPe\_STNdm\_W": 1.5,**

&#x20;       **"GPe\_GPi\_SNpr\_W": 0.1**

&#x20;   **},**

&#x20;   **"BG\_v\_W": {**

&#x20;       **"NAc\_1\_SNpr\_W": 3.0,**

&#x20;       **"NAc\_2\_GPe\_W": 2.5,**

&#x20;       **"STNv\_SNpr\_W": 3.0,**

&#x20;       **"STNv\_GPe\_W": 2.0,**

&#x20;       **"GPe\_STNv\_W": 1.5,**

&#x20;       **"GPe\_SNpr\_W": 0.1**

&#x20;   **},**

&#x20;   **"SNpc\_W": {**

&#x20;       **"SNpci\_1\_SNpco\_1\_W": 0.52,**

&#x20;       **"SNpci\_2\_SNpco\_2\_W": 0.52**

&#x20;   **},**

&#x20;   **"seed": 0,**

&#x20;   **"noise": {**

&#x20;       **"BG\_dl": 0.0,**

&#x20;       **"BG\_dm": 0.0,**


params.py
"""
Created on Sat Aug 16 11:50:44 2025

@author: Utente
"""

from parameter_manager import ParameterManager


class Parameters(ParameterManager):
    def __init__(
        self,
        seed=0,
        N={
            "BG_dl": 2,
            "BG_dm": 2,
            "BG_v": 2,
            "MGV": 2,
            "MC": 2,
            "BLA_IC": 4,
            "SNpc": 2,
            "PPN": 1,
            "LH": 1,
            "VTA": 1,
            "P": 2,
            "DM": 2,
            "PL": 2,
            "PFCd_PPC": 2,
        },
        threshold={
            "BG_dl": 0.0,
            "BG_dm": 0.0,
            "BG_v": 0.0,
            "MGV": 0.0,
            "MC": 0.2,
            "BLA_IC": 0.0,
            "SNpc": 0.0,
            "PPN": 0.0,
            "LH": 0.0,
            "VTA": 0.0,
            "P": 0.0,
            "DM": 0.0,
            "PL": 0.0,
            "PFCd_PPC": 0.0,
        },
        tau={
            "MC": 6,
            "PFCd_PPC": 6,
            "PL": 6,
            "MGV": 6,
            "P": 6,
            "DM": 6,
            "BG_dl": 6,
            "BG_dm": 6,
            "BG_v": 6,
            "BLA_IC": [10, 10],
            "SNpc_i": 6,
            "SNpc_o": 6,
            "PPN": [2, 10],
            "LH": [2, 10],
            "VTA": 6,
        },
        baseline={
            "PPN": 0.0,
            "LH": 0.0,
            "VTA": 0.0,
            "BLA_IC": 0.0,
            "SNpc": 0.7,
            "DLS": 0.0,
            "STNdl": 0.0,
            "GPi": 0.4,
            "DMS": 0.0,
            "STNdm": 0.0,
            "GPi_SNpr": 0.4,
            "NAc": 0.0,
            "STNv": 0.0,
            "SNpr": 0.4,
            "MGV": 0.1,
            "MC": 0.0,
            "P": 0.1,
            "PFCd_PPC": 0.0,
            "DM": 0.1,
            "PL": 0.0,
        },
        BG_dl_W={"DLS_GPi_W": 1.8, "STNdl_GPi_W": 1.4},
        BG_dm_W={"DMS_GPiSNpr_W": 1.8, "STNdm_GPiSNpr_W": 1.4},
        BG_v_W={"NAc_SNpr_W": 1.8, "STNv_SNpr_W": 1.4},
        SNpc_W={"SNpci_1_SNpco_1_W": 1.0, "SNpci_2_SNpco_2_W": 1.0},
        noise={
            "BG_dl": 0.0,
            "BG_dm": 0.0,
            "BG_v": 0.0,
            "MGV": 0.0,
            "MC": 0.2,
            "BLA_IC": 0.0,
            "SNpc": 0.0,
            "PPN": 0.0,
            "LH": 0.0,
            "VTA": 0.0,
            "P": 0.0,
            "DM": 0.0,
            "PL": 0.2,
            "PFCd_PPC": 0.2,
        },
        Matrices_scalars={
            "Mani_DLS": 0.0,
            "Mani_DMS": 0.0,
            "Mani_BLA_IC": 5.0,
            "Food_PPN": 10.0,
            "Food_BLA_IC": 5.0,
            "Food_LH": 10.0,
            "Sat_BLA_IC": 10.0,
            "PPN_SNpco": 20.0,
            "BLA_IC_NAc": 0.0,
            "BLA_IC_LH": 5.0,
            "LH_VTA": 20.0,
            "NAc_SNpci_1": 6.0,
            "DMS_SNpci_2": 10.0,
            "GPi_MGV": 1.5,
            "GPi_SNpr_P": 1.5,
            "SNpr_DM": 1.5,
            "MGV_MC": 1.0,
            "P_PFCd_PPC": 1.0,
            "DM_PL": 1.0,
            "PL_DM": 1.0,
            "PL_NAc": 1.2,
            "PL_STNv": 1.6,
            "PL_PFCd_PPC": 0.2,
            "PFCd_PPC_P": 1.0,
            "PFCd_PPC_DMS": 1.2,
            "PFCd_PPC_STNdm": 1.6,
            "PFCd_PPC_PL": 1.0,
            "PFCd_PPC_MC": 0.3,
            "MC_MGV": 1.0,
            "MC_DLS": 1.2,
            "MC_STNdl": 1.6,
            "MC_PFCd_PPC": 0.2,
        },
        DA_values={
            "Y_DLS": 0.2,
            "Y_DMS": 0.5,
            "Y_NAc": 0.8,
            "delta_DLS": 4.0,
            "delta_DMS": 6.5,
            "delta_NAc": 1.5,
        },
        BLA_Learn={"eta_b": 0.08, "alpha_t": 50.0, "tau_t": 200, "theta_DA": 0.7, "max_W": 2},
        Str_Learn={
            "eta_DLS": 0.02,
            "eta_DMS": 0.02,
            "eta_NAc": 0.05,
            "theta_DA_DLS": 0.3,
            "theta_DA_DMS": 0.3,
            "theta_DA_NAc": 0.5,
            "theta_DLS": 0.5,
            "theta_DMS": 0.5,
            "theta_NAc": 0.5,
            "theta_inp_DLS": 0.5,
            "theta_inp_DMS": 0.5,
            "theta_inp_NAc": 0.9,
            "max_W_DLS": 1,
            "max_W_DMS": 1,
            "max_W_NAc": 2,
        },
    ):

        self.N = N
        self.threshold = threshold
        self.tau = tau
        self.baseline = baseline
        self.BG_dl_W = BG_dl_W
        self.BG_dm_W = BG_dm_W
        self.BG_v_W = BG_v_W
        self.SNpc_W = SNpc_W
        self.seed = seed
        self.noise = noise
        self.Matrices_scalars = Matrices_scalars
        self.DA_values = DA_values
        self.BLA_Learn = BLA_Learn
        self.Str_Learn = Str_Learn

        super(Parameters, self).__init__()


# param_string = "Simulation"
# param_file = "prm_file"

# parameters = Parameters()
# parameters.update(param_string)
# parameters.save(param_file + ".json", mode="json")
